### PR TITLE
Handle missing optional keys

### LINKED ISSUE #49: ZGW Consumers can't handle missing values, even if they're optional
As discussed zgw-consumers errors when a ZGW server doesn't return all the expected fields in the api model:

> missing 9 required positional arguments: 'omschrijving_generiek', 'toelichting', 'trefwoorden', 'publicatietekst', 'verantwoordingsrelatie', 'eigenschappen', 'deelzaaktypen', 'einde_geldigheid', and 'concept'

This came up in OIP: https://taiga.maykinmedia.nl/project/open-inwoner/issue/773
and is patched in OIP using derived dataclasses: https://github.com/maykinmedia/open-inwoner/pull/305/files

As mentioned in Slack on today / 5 sept:

> volgens de semantics van OAS 3 is het niet verplicht om niet-required velden terug te geven
> beste route hier is om zgw-consumers te patchen met een default factory voor optionele velden, lijkt me

A broader issue is that DRF translates blank=False to required=True in the OAS

### PR DESCRIPTION
**Pull request by sergei-maertens**
Fixes #49

@alextreme - is there any chance you could try out this branch/fix in OIP before we merge it?

### CODE REVIEW

**Comment by alextreme** [comment]
Unfortunately I wasn't easily able to switch zgw_consumers out, we're on 0.18 and switching to 0.21 gave me sufficient simple_certmanager-errors that I didn't want to do this live on acceptance.

I did the next best thing and added a few tests based on the OIP acceptance data:

https://github.com/maykinmedia/zgw-consumers/pull/53

It seems the other party isn't returning sufficient mandatory data for the ZaakTypes and Zaak.

**Comment by sergei-maertens** [comment]
Thanks, I merged that and will look into it further.

**Comment by alextreme** [comment]
Umm that wasn't my intention, I suspect that if you do look into it that you'll want to remove these 2 tests :)

File: tests/api_models/test_optional_fields.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        "indicatieInternOfExtern": "intern",
+        "omschrijving": "Aangeven",
+        "onderwerp": "Aangeven",
+        "url": "https://uniek-url",
+        "vertrouwelijkheidaanduiding": "openbaar",
+    }
+    instance = factory(ZaakType, data)
+
+    assert instance.omschrijving == "Aangeven"
+    assert instance.omschrijving_generiek is None
```
Comment (sergei-maertens) [inline tests/api_models/test_optional_fields.py:40]: Looked into this failing test and compared with the [API spec](https://catalogi-api.vng.cloud/api/v1/schema/#operation/zaaktype_read):

* `catalogus`is a required string field (format uri)
* `doorlooptijd` is a required string field (format duration)
* `opschorting_en_aanhouding_mogelijk` is a required boolean field
* `verlenging_mogelijk` is a required boolean field
* `publicatie_indicatie` is a required boolean field
* `producten_of_diensten` is a required list of strings (format uri)
* `besluittypen` is a required list of strings (format uri)
* `begin_geldigheid` is a required string field (format date)
* `versiedatum`  is a required string field (format date)

In short: the API provider is not standard-compliant here.

This test is useful, but should be adapted `with pytest.assertRaises(TypeError): ...` instead of the asserts.

File: tests/api_models/test_optional_fields.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        "startdatum": "2022-07-28",
+        "url": "https://uniek-url",
+        "uuid": "53d12522-6cb5-583e-b6f6-7c17a2fcd440",
+        "verantwoordelijkeOrganisatie": "CHANGEME",
+        "zaaktype": "https://uniek-url",
+    }
+    instance = factory(Zaak, data)
+
+    assert instance.bronorganisatie == "CHANGEME"
+    assert instance.toelichting is None
```
Comment (sergei-maertens) [inline tests/api_models/test_optional_fields.py:59]: This one is missing `vertrouwelijkheidaanduiding`, which is a bug in the standard. The business logic is that the VA is derived from the zaaktype if not explicitly provided - so it is definitely intended to be set and communicated throughout the API.

This is also why the field is not `required` in the API schema - it is not for the WRITE operations, but it is for the READ operations (that's the bug).

**Comment by sergei-maertens** [comment]
> Umm that wasn't my intention, I suspect that if you do look into it that you'll want to remove these 2 tests :)

No worries, this makes it easy to check out and everything can go away through the power of rebase ;)
